ambry-store/src/test/java/com.github.ambry.store/CompactionManagerTest.java

```diff
@@ -229,5 +366,35 @@ long getCapacityInBytes() {
     List<String> getLogSegmentsNotInJournal() throws StoreException {
       return validLogSegments;
     }
+
+    @Override
+    void compact(CompactionDetails details) {
+      compactCalled = true;
+      compactCallsCountdown.countDown();
+      if (details == null) {
+        callOrderException = new Exception("Called compact() with null details");
+      } else if (!resumeCompactionCalled) {
+        callOrderException = new Exception("Called compact() before calling maybeResumeCompaction()");
+      }
+      if (exceptionToThrowOnCompact != null) {
+        throw exceptionToThrowOnCompact;
+      }
+    }
+
+    @Override
+    void maybeResumeCompaction() {
+      if (resumeCompactionCalled) {
+        callOrderException = new Exception("maybeResumeCompaction() called more than once");
```

Comment:
Similar reply - not intended to mimic store code, but intended to test the flow of compaction thread - which should not call `resumeCompaction()` for a store more than once in one lifetime.